### PR TITLE
add configuration support for gRPC TLS

### PR DESCRIPTION
**Pull request by maxthomas**
this adds support for gRPC TLS in the configuration and in the application.

i think there's an issue in the current `FromYAML` method in that it doesn't actually read the `grpc_address` field, it just uses the one from `defaultApp`. so i'm not sure if that's worth fixing alongside this.

anyway, let me know if you have any feedback, i'd also be happy to add HTTPS security if you'd like to see that before merging.

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -542,3 +555,38 @@ func getDockerCID() (string, error) {
 type proxyProb struct {
 	Proxies yaml.MapSlice
 }
+
+type AppTLS struct {
```
Comment (horkhe) [inline config/config.go]: I would have named it just `TLS` and put `cert_path` and `key_path` directly under it. If TLS is configured then both gRPC and HTTP interface should use it. It is ok if this PR only enables them for gRPC, later we can make HTTP use it too. If feel like implementing it right here that would be great.

Comment (maxthomas) [reply config/config.go]: sounds good, i'll push up the changes for that as well

Comment (maxthomas) [reply config/config.go]: :ping_pong: with feedback. the changes to the proxy are a little ugly, but it's not too bad, imo

**Comment by horkhe** [comment]
@thrawn01 do you have anything to add? The code of `FromYAML` looks weird because my original logic only loaded the Proxies part correctly, and the rest was set to default. #168 was created to track a fix for that.

**Comment by horkhe** [comment]
One more thing, if you could update the README and CHANGELOG files with information about support for TLS that would be great.

**Comment by maxthomas** [comment]
wasn't sure if it was an implemented or fixed, so put implemented and also tagged issue. let me know what you think

**Pull request merged**